django/test/runner.py

```diff
@@ -101,6 +101,43 @@ def debug(self, error):
         pdb.post_mortem(traceback)
 
 
+class TestCaseDTO:
+    """
+    The ParallelTestSuite requires that all test cases be pickleable. When a
+    SubTest fails with an exception, it's pickled and shipped back to the main
+    process to be displayed to the user. Unfortunately, Django's TestCases
+    aren't always pickleable (for example, they may contain an instance of
+    django.test.Client, which can't be cleanly pickled if an exception was
+    raised by the view).
```

Comment:
If `TestCase`s aren't pikcleable, wouldn't that affect normal tests, and not just subtests too?

Reply:
The `RemoteTestResult.events` list is pickled on its way back from `django.test.runner._run_subsuite`. '

When dealing with normal tests, the data in `events` just consists of test indices and possibly errors, and the main process uses those indices to retrieve the tests.

However, when a subtest is encountered,`django.test.runner.RemoteTestResult.addSubTest` puts the actual subtest instance into the `events` list. This is because [`unittest.TestResult.addSubTest`](https://docs.python.org/3/library/unittest.html#unittest.TestResult.addSubTest) expects to receive the subtest instance.